test/FunctionalTests/Linker/LinkerTests.cs

```diff
@@ -110,7 +133,9 @@ private static async Task PublishAppAsync(string path, string outputPath, Cancel
 
         try
         {
-            process.Start($"publish {resolvedPath} -r {GetRuntimeIdentifier()} -c Release -o {outputPath} --self-contained");
+            // The AppPublishAot parameter is used to tell the compiler to publish as AOT.
+            // AppPublishAot is used instead of PublishAot because dependency projects have non-AOT targets. Setting "PublishAot=true" causes build errors.
+            process.Start("dotnet", $"publish {resolvedPath} -r {GetRuntimeIdentifier()} -c Release -o {outputPath} -p:AppPublishAot={publishAot} --self-contained");
```

Comment:
Does this need to add ".exe" for windows?

Reply:
TIL it's optional. Remove logic to add *.exe and added a comment explaining why.